algebra/ring.lean

```diff
@@ -15,6 +15,85 @@ section
   (left_distrib n 1 1).trans (by simp)
 end
 
+section units
+  variables [semiring α]
+  variable (α)
+
+  structure invertible :=
+    (val : α)
+    (inv : α)
+    (property : val * inv = 1 ∧ inv * val = 1)
+
+  variables {a b c : invertible α}
+  variable {α}
+
+  def ext (HAB : a.val = b.val) : a = b :=
```

Comment:
Did you just create a theorem `ext` in the root namespace? That's obviously going to cause problems. Probably most of the theorems here should be in the `units` namespace.